packages/chakra-components/src/ComboBox/useSelect.js

```diff
@@ -164,7 +167,7 @@ export default function useSelect({
   // If multi and duplicates aren't allowed, filter out the
   // selected options from the options list
   options = React.useMemo(() => {
-    if (multi && !duplicates && !async) {
+    if (multi && !duplicates && !async && !isListBox) {
```

Comment:
We don't have duplicates right ?

Reply:
Currently I am only focused on MultiSelect folder which has all the behaviours.
These changes appeared in my PR because of rebase with the original branch after you merged your ListBox version.